### PR TITLE
Bring changes from dec_2023_release branch into master branch

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Description
This brings in all the changes from dec_2023_release branch to master branch for preparing for December 2023 release.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - presubmit

### CODE REVIEW

File: tools/cd_scripts/install_test.sh
Comment (gargnitingoogle) [inline tools/cd_scripts/install_test.sh:33]: @ashmeenkaur @sethiay please check this file. This had a lot of conflicts on merging between the following commits.

* https://github.com/GoogleCloudPlatform/gcsfuse/commit/70ed0ef3f47d605de3354b3652b77617b3bc26ea
* https://github.com/GoogleCloudPlatform/gcsfuse/commit/a510793b68715877c95d77f1c06bd406a638b292

Comment (gargnitingoogle) [reply tools/cd_scripts/install_test.sh:33]: @Tulsishah  for change in arm-related changes.

**Comment by gargnitingoogle** [comment]
@sethiay @ashmeenkaur @Tulsishah please review this PR for your changes.

**Review by ashmeenkaur — APPROVED** [review]

File: tools/cd_scripts/install_test.sh
Comment (Tulsishah) [reply tools/cd_scripts/install_test.sh:33]: Please go with @ashmeenkaur latest changes. I checked looks good to me.

**Review by Tulsishah — DISMISSED** [review]

File: tools/cd_scripts/install_test.sh
Comment (ashmeenkaur) [reply tools/cd_scripts/install_test.sh:33]: Please accept the changes from release branch.

File: tools/cd_scripts/install_test.sh
Comment (gargnitingoogle) [inline tools/cd_scripts/install_test.sh:72]: I'm not sure what's going on with this bit here and in other places.

Comment (gargnitingoogle) [reply tools/cd_scripts/install_test.sh:54]: Done.

File: tools/cd_scripts/install_test.sh
Comment (gargnitingoogle) [reply tools/cd_scripts/install_test.sh:20]: > Please accept the changes from release branch.

Done.

Also, I realized where most of the conflicts were from.
1. master: https://github.com/GoogleCloudPlatform/gcsfuse/commit/70ed0ef3f47d605de3354b3652b77617b3bc26ea
2. dec_2023_release: https://github.com/GoogleCloudPlatform/gcsfuse/commit/dc6aecaed2fbb326b6415f165cb5b4c173465e63

**Comment by gargnitingoogle** [comment]
Integration tests an perf presubmit passed: [log](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/435d885e-a009-4abd-9287-f797ca30e590/log)

**Comment by gargnitingoogle** [comment]
@ashmeenkaur @sethiay made the change you asked for. please approve.

**Comment by sethiay** [comment]
> @ashmeenkaur @sethiay made the change you asked for. please approve.

Changes wise looks good to me. @ashmeenkaur please approve after running it on CD pipeline (install test only).

**Review by ashmeenkaur — APPROVED** [review]

**Comment by gargnitingoogle** [comment]
@sethiay please approve.

**Review by sethiay — APPROVED** [review]
Please also get approval from @raj-prince

**Comment by gargnitingoogle** [comment]
@raj-prince please review and approve.

Also, after approving, please rebase and merge this PR, and then run the long running tests, and update here.

**Review by raj-prince — APPROVED** [review]

**Comment by raj-prince** [comment]
Started the long haul test (tf, pytorch_v2, pytorch_v1_2).

**Pull request merged**
